### PR TITLE
Improvements

### LINKED ISSUE #6882: timeline refactoring
- [x] support complicated cases e.g. https://triage.cbioportal.mskcc.org/patient?caseId=P-0000012&studyId=mskimpact_test_june
- [x] bottom circles shade with multiple events are cut off
- [x] download timeline as table from clinical tab
- [x] Trimming does not work when browser is zoomed in or zoomed out

### PR DESCRIPTION
**Pull request by Luke-Sikina**
# What? Why?
Fix cBioPortal/cbioportal#6882.

Changes proposed in this pull request:
- Zoom + Trim
  - Issue: when trimming the timeline, zooming would zoom on the
  wrong region
  - Root cause: start and end of zoom region were determined
  according to untrimmed timeline. This was happening inside the
  d3 library itself
  - Fix: take the point the the brush generates, determine its
  placement on the untrimmed timeline, find the nearest tick
  on the trimmed timeline, replace the point with that.
- Cut off circles
  - Issue: mousing over points on the bottom lane of the
  timeline caused them to be cut off when they expanded.
  - Fix: added 3 pixels to the height of the svg

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).
zoom

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,14 @@ Checks before sending a PR:
 - Run all tests to assure nothing else was accidentally broken. This is done by running: `npm test`.
 - Make sure you have added the necessary tests and [documentation](http://usejsdoc.org/) for your changes.
 
+## Development
```
Comment (Luke-Sikina) [inline README.md:42]: It took me a minute to figure out how to do dev, so I added this to help people. Please correct me if there's a faster way to iterate.

File: js/plugins/zoom.js [hunk 2]
```diff
@@ -37,14 +74,16 @@ clinicalTimelineZoom.prototype.run = function(timeline, spec) {
      */
     var brushend = function() {
       var originalZoomLevel = timeline.computeZoomLevel(minDays, maxDays, width);
+      const brushStart = adjustLeftEdge(timeline, brush.extent()[0].valueOf(), minDays, maxDays, originalZoomLevel);
+      console.log("Brush start: ", brushStart);
```
Comment (Luke-Sikina) [inline js/plugins/zoom.js]: delete

File: README.md (hunk 1 above)
Comment (alisman) [reply README.md:42]: nice!

**Review by inodb — APPROVED** [review]

**Comment by Luke-Sikina** [comment]
Redid almost everything I changed in this PR, felt this was best expressed in a new PR.

**Pull request closed without merging**